processor/schemaprocessor/internal/translation/revision_v1.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"go.opentelemetry.io/otel/schema/v1.0/ast"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/migrate"
+)
+
+// RevisionV1 represents all changes that are to be
+// applied to a signal at a given version.
+type RevisionV1 struct {
+	ver              *Version
+	all              *migrate.AttributeChangeSetSlice
+	resource         *migrate.AttributeChangeSetSlice
+	spans            *migrate.ConditionalAttributeSetSlice
+	eventNames       *migrate.SignalSlice
+	eventAttrsOnSpan *migrate.ConditionalAttributeSetSlice
+	eventAttrsOnName *migrate.ConditionalAttributeSetSlice
+	metricsAttrs     *migrate.ConditionalAttributeSetSlice
+	metricNames      *migrate.SignalSlice
+}
+
+// NewRevision processes the VersionDef and assigns the version to this revision
+// to allow sorting within a slice.
+// Since VersionDef uses custom types for various definitions, it isn't possible
+// to cast those values into the primitives so each has to be processed together.
+// Generics would be handy here.
+func NewRevision(ver *Version, def ast.VersionDef) *RevisionV1 {
+	return &RevisionV1{
+		ver:              ver,
+		all:              newAttributeChangeSetSliceFromChanges(def.All),
+		resource:         newAttributeChangeSetSliceFromChanges(def.Resources),
+		spans:            newSpanConditionalAttributeSlice(def.Spans),
+		eventNames:       newSpanEventSignalSlice(def.SpanEvents),
+		eventAttrsOnSpan: newSpanEventConditionalSpans(def.SpanEvents),
+		eventAttrsOnName: newSpanEventConditionalNames(def.SpanEvents),
+		metricsAttrs:     newMetricConditionalSlice(def.Metrics),
+		metricNames:      newMetricNameSignalSlice(def.Metrics),
+	}
+}
+
+func newAttributeChangeSetSliceFromChanges(attrs ast.Attributes) *migrate.AttributeChangeSetSlice {
+	values := make([]*migrate.AttributeChangeSet, 0, 10)
```

Comment:
Any particular reason to use 10 as capacity?

Reply:
It was just a random number I pulled from no where.

Reply:
Nit: since it has no particular importance I think it can be safely deleted, `values := make([]*migrate.AttributeChangeSet)` should be good enough.

Reply:
Unfortunately, (not sure when), but the latest version of go requires at least `len` and an optional `cap`.

Tested it here (to make sure it wasn't a linter getting it in the way)
https://go.dev/play/p/9uNpeSYw6gN

Reply:
You are right. Don't know what I was thinking. What I probably wanted to say is that you can start with an empty slice since `append` later will happily allocate as many as needed:
```go
var values []*migrate.AttributeChangeSet
```